### PR TITLE
Multilayer Perceptron model

### PR DESCRIPTION
**Pull request by anandsaini024**
Training a multilayer perceptron.

### CODE REVIEW

**Comment by anandsaini024** [comment]
There is some problem in the training as the Accuracy is always exactly same.

**Comment by adam2392** [comment]
Looks like a good start! 

Once you figure out the issue w/ training, let us know. First, try to overfit the data if you can and then we can work backwards: e.g. get 100% training accuracy so we at least know it's working

**Comment by anandsaini024** [comment]
> Looks like a good start!
> 
> Once you figure out the issue w/ training, let us know. First, try to overfit the data if you can and then we can work backwards: e.g. get 100% training accuracy so we at least know it's working

I have made some changes, it seems to be working now.

**Comment by adam2392** [comment]
Yeah the code looks good. I think what would be helpful is instead of code, perhaps iterating on some dataset and results that we expect to see. 

I guess the goal here is to i) validate the source localization feature and ii) propose a base NN model that you want to iterate on against ICLabel.

Re i), you should maybe create some figures or output that we can then analyze to determine if this feature is "doing what we want it to do". Idk what that would be tho since I'm not a source localization expert.

Re ii), this is just your traditional compare/contrast models on the same train/test data.

Wdyt @mscheltienne ?

**Comment by anandsaini024** [comment]
I have one doubt with this:
The cost/loss is not decreasing throughout the epochs, it's almost same for all the epochs. I am using Binary cross-entropy loss. Am I using wrong loss function?
I tried increasing the number of epochs to 3000 but only marginal decrease in the cost.
Then, I tried increasing the layers and neurons but the accuracy decreased with that.
I think there is still some problem in it, can you guys run it once and see?
(If you don't have mara data in your local, please let me know I have script to get the mara data with labels in a proper format and then you can simply edit the path to data.)


Epoch: 100/100 | Batch 001/022 | Cost: 0.6120,
Epoch: 100/100 | Batch 002/022 | Cost: 0.6331,
Epoch: 100/100 | Batch 003/022 | Cost: 0.6061,
Epoch: 100/100 | Batch 004/022 | Cost: 0.6612,
Epoch: 100/100 | Batch 005/022 | Cost: 0.7702,
Epoch: 100/100 | Batch 006/022 | Cost: 0.5913,
Epoch: 100/100 | Batch 007/022 | Cost: 0.5770,
Epoch: 100/100 | Batch 008/022 | Cost: 0.5792,
Epoch: 100/100 | Batch 009/022 | Cost: 0.7237,
Epoch: 100/100 | Batch 010/022 | Cost: 0.6863,
Epoch: 100/100 | Batch 011/022 | Cost: 0.6797,
Epoch: 100/100 | Batch 012/022 | Cost: 0.7168,
Epoch: 100/100 | Batch 013/022 | Cost: 0.6336,
Epoch: 100/100 | Batch 014/022 | Cost: 0.6170,
Epoch: 100/100 | Batch 015/022 | Cost: 0.6856,
Epoch: 100/100 | Batch 016/022 | Cost: 0.6712,
Epoch: 100/100 | Batch 017/022 | Cost: 0.7137,
Epoch: 100/100 | Batch 018/022 | Cost: 0.7093,
Epoch: 100/100 | Batch 019/022 | Cost: 0.6835,
Epoch: 100/100 | Batch 020/022 | Cost: 0.6861,
Epoch: 100/100 | Batch 021/022 | Cost: 0.6561,
Epoch: 100/100 | Batch 022/022 | Cost: 0.6579,
Epoch: 100/100 Train Cost: 0.7020

Training Accuracy: 55.56
Test Loss: 0.02643927527062687
Test_accuracy: 57.407407407407405

**Comment by anandsaini024** [comment]
Okay, so now it's working. 100% training accuracy. PLease let me know if you have any suggestions on different models we can try.

**Comment by adam2392** [comment]
Nice! What was the issue?

**Comment by anandsaini024** [comment]
> Nice! What was the issue?

I was not using the activation layer in between the layers so it was acting as one layer and hence was not learning.
